.github/workflows/terraform.yml

```diff
@@ -0,0 +1,38 @@
+name: "terraform"
+
+# for security should only run on push to bors branches (staging/trying)
+on:
+  push:
+    branches:
+      - staging
+      - trying
+
+jobs:
+  terraform-deploy:
+    if: github.repository == 'nix-community/infra'
+    runs-on: ubuntu-latest
+    env:
+      # TODO: add to repo secrets
+      SOPS_AGE_KEY: ${{ secrets.SOPS_AGE_KEY }}
+      TF_IN_AUTOMATION: true
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cachix/install-nix-action@v18
+        with:
+          extra_nix_config: |
+            experimental-features = nix-command flakes
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+      - name: init
+        run: nix develop -c ./terraform/deploy terraform init -input=false
+      - name: validate
+        run: nix develop -c ./terraform/deploy terraform validate
+      # TODO: fmt files
+      #- name: fmt
+      #  run: nix develop -c ./terraform/deploy terraform fmt -check
+      - name: plan
+        if: github.ref == 'refs/heads/trying'
+        run: nix develop -c ./terraform/deploy terraform plan -input=false
+      - name: apply
+        # master == remote state
+        if: github.ref == 'refs/heads/staging'
+        run: nix develop -c ./terraform/deploy terraform apply -auto-approve -input=false
```

Comment:
For more fancy approaches, see https://github.com/hashicorp/setup-terraform#usage

It would be nice to display the planned changes on the PR before merging.

Reply:
Oops, I had typed a response for this but mustn't have clicked the button.

`bors` will always post a comment with a link to the CI run, two comments on every run may be a bit confusing or noisy. Also posting comments on PRs is slightly complicated by the bors workflow as it's a push event, not a pull request so we'd also need to tell it which PR it originated from.